test/test-coordinator.js

```diff
@@ -95,42 +95,43 @@ require('util').inherits(TestCoordinator, events.EventEmitter);
 
 TestCoordinator.prototype.lookup = function(key) {
     var self = this;
-
-    var matchingNode;
-    var matchingHash;
-
     var hash = farmhash.fingerprint32(key);
 
-    function isBetterMatch(newHash) {
-        if (!matchingNode) return true; // 1 node is better than no node
-
-        if (hash <= newHash && newHash < matchingHash) return true; // new hash is closer to the current best match
-        if (matchingHash <= hash && hash <= newHash) return true; // the matchingHash wraps around, and new hash is a better match to hash
-
-        return false;
-    }
-
     var hostPorts = this.fakeNodes.map(function (fn) {
         return fn.getHostPort();
     });
     hostPorts.push(this.sutHostPort);
 
     // iterate over all hosts
+    var allHashes = [];
```

Comment:
I found this hard to follow but maybe I'm just looking at it differently.

Can't we just do something like:
```
var min = Infinity;
var node = undefined;

hostPorts.forEach(function (hostPort) {
  for (var i=0; i<self.replicaPoints; i++) {
    var replica = farmhash.fingerprint32(hostPort + i);
    var dist = Math.min(replica - hash, replica-hash + Math.pow(2, 32))
    if (dist < min) {
        min = dist;
        node = hostPort;
    }
  }
}
```

Again, I could just be looking at it differently miss a subtlety..